org.eclipse.xtext.tests/xtend-gen/org/eclipse/xtext/xtext/ecoreInference/Xtext2EcoreTransformerTest.java

```diff
@@ -2091,6 +2095,33 @@ public void testBug296496_02() throws Exception {
       this.<EStructuralFeature>feature(createdType, "otherEnumFeature").getEType());
   }
   
+  @Test
+  public void testEclipseIssue1547() throws Exception {
+    StringConcatenation _builder = new StringConcatenation();
+    _builder.append("grammar test with org.eclipse.xtext.common.Terminals");
+    _builder.newLine();
+    _builder.append("generate myDsl \'uri\'");
+    _builder.newLine();
+    _builder.append("Model: element=Element;");
+    _builder.newLine();
+    _builder.append("enum Element : ^false=\'false\' | ^true=\'true\';");
```

Comment:
Quoting shouldn't be necessary anymore given that we call ValidID now.

Reply:
For other IDs it is necessary though.

Reply:
?

Reply:
ok will test both.